include/saber/handler.hpp

```diff
@@ -80,6 +81,68 @@ class ValueHandler
 
 }; // class ValueHandler
 
+template<typename T>
+class ReferenceHandler
+{
+public:
+    // ctor stores a copy of the variable's stored data
+    ReferenceHandler(T* inReference) :
+        mReference{inReference}
+    {
+        // Do nothing
+    }
+
+    ReferenceHandler() = default;
+
+    ~ReferenceHandler() = default;
+
+    // Move Ctor
+    /// @brief Construct this with data moved from inputted ReferenceHandler
+    /// @param ioMove ReferenceHandler which will have its data swapped with this, nullifying it 
+    ReferenceHandler(ReferenceHandler&& ioMove) noexcept
+    {
+        if (ioMove)
+        {
+            // swap allows for noexcept move ctor
+            std::swap(mReference, ioMove.mReference);
+        }
+    }
+
+    // Move Operator
+    /// @brief Assign this with data moved from inputted ReferenceHandler
+    /// @param ioMove ReferenceHandler which will have its data swapped with this
+    /// @return this
+    ReferenceHandler& operator=(ReferenceHandler&& ioMove) noexcept
+    {
+        // Moving to self should be a NOP
+        if (this != &ioMove)
+        {
+            // swap allows for noexcept move assign
+            std::swap(mReference, ioMove.mReference);
+        }
+        return *this;
+    }
+
+    operator bool()
+    {
+        return (mReference != nullptr);
+    }
+
+    T* Get()
+    {
+        return mReference.get();
+    }
+
+    void Reset() noexcept
+    {
+        mReference.reset();
+    }
+
+private:
+    std::unique_ptr<T> mReference{nullptr};
```

Comment:
You'll need to add you nested `class Deleter` right around here...
Similar to what you did with `ControlBlock` for `shared_ptr`.
It will be a functor class just like `std::default_delete<>`, so you could use/copy that for inspiration:
https://en.cppreference.com/w/cpp/memory/default_delete